tests/drivers/clock_control/clock_control_mpsl/src/test_clock_control.c

```diff
@@ -9,7 +9,7 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(test);
 
-#if DT_HAS_NODE(DT_INST(0, nordic_nrf_clock))
+#if DT_NODE_EXISTS(DT_INST(0, nordic_nrf_clock))
```

Comment:
I think this and the below `#if` statements should simply be removed. Perhaps they are copied from upstream? I am unaware of any Nordic SoC that doesn't have this peripheral and require it to be available in applications, but I may be missing something here.

If the checks need to stay, they should be changed to DT_HAS_COMPAT_STATUS_OKAY(nordic_nrf_clock).

It is *not* safe to replace the now-removed DT_HAS_NODE with DT_NODE_EXISTS. In general, DT_HAS_NODE(foo) can be safely replaced with DT_NODE_HAS_STATUS(foo, okay), but I think that's not  the best change in this particular case.

The main DTS related change in this merge is that *all* node data is available now, where formerly only nodes that were enabled (status = "okay" or no status property set) could be accessed from C. So the old DT_HAS_NODE() actually was doing double duty checking node existence and status, and that's why it was split into DT_NODE_EXISTS() and DT_NODE_HAS_STATUS(), which each do one job.

In this case, just checking if the compatible is enabled is better, because using instance numbers in this way can lead to subtle bugs so it's best avoided as a general practice, though you can probably get away with it here. This comment is already long so please ask if you want more details.